ideal-image-slider.js

```diff
@@ -440,11 +441,35 @@ var IdealImageSlider = (function() {
 		}
 
 		// Slider (container) element
-		var sliderEl = document.querySelector(this.settings.selector);
-		if (!sliderEl) return null;
+		var sliderEl = false,
+			rquickExpr =  /^(?:#([\w-]+)|(\w+)|\.([\w-]+))$/,
+			selectionType = rquickExpr.exec( this.settings.selector ),
+			s, elem;
+		
+		// Handle ID, Name, and Class selection
+		if ((s = selectionType[1])) {
+			// getElementById can match elements by name instead of ID
+			if ((elem = document.getElementById( s )) && elem.id === s)
+				sliderEl = elem;
+		} else if (selectionType[2]) {
+			if ((elem = document.getElementsByTagName( this.settings.selector )))
+				sliderEl = elem[0];
+		} else if ((s = selectionType[1])) {
+			if ((elem = document.getElementsByClassName( s )))
+				sliderEl = elem[0];
+		} else {
+			sliderEl = document.querySelector(this.settings.selector);
+		}
+
+		if (!sliderEl) {
+			return null;
+		}
+
+		//Add loading class
+		_addClass(sliderEl, 'iis-loading');
```

Comment:
In the future I'll adhere to a vary feature independent PR style. In the name of closing this PR, can we just keep the loading classes in for now?